src/utils.rs

```diff
@@ -22,31 +40,41 @@ pub(crate) fn map_path_to_uri(uri_or_sha_prefix: &str) -> Result<String> {
         if let Some(policy) = store.get_policy_by_sha_prefix(uri_or_sha_prefix)? {
             Ok(policy.uri.clone())
         } else {
-            Err(anyhow!(
-                "Cannot find policy with prefix: {}",
-                uri_or_sha_prefix
-            ))
+            Err(LookupError::PolicyMissing(uri_or_sha_prefix.to_string()))
         }
     }
 }
 
-pub(crate) fn wasm_path(uri: &str) -> Result<PathBuf> {
+pub(crate) fn get_uri(uri_or_sha_prefix: &String) -> std::result::Result<String, LookupError> {
+    map_path_to_uri(uri_or_sha_prefix).or_else(|_| {
+        Reference::from_str(uri_or_sha_prefix)
+            .map(|oci_reference| format!("registry://{}", oci_reference.whole()))
+            .map_err(|_| LookupError::PolicyMissing(uri_or_sha_prefix.to_string()))
+    })
+}
+
+pub(crate) fn get_wasm_path(uri_or_sha_prefix: &str) -> std::result::Result<PathBuf, LookupError> {
+    let uri = crate::utils::get_uri(&uri_or_sha_prefix.to_owned())?;
+    crate::utils::wasm_path(&uri)
```

Comment:
it's the same crate, we don't need the full path `crate::utils::`

Reply:
Thanks, copy/paste issue. xD